### ISSUE TITLE
Issue 005: overlap 付き channel tile の coarse 予測を gather 座標へ統合する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#320]
# Issue 005: overlap 付き channel tile の coarse 予測を gather 座標へ統合する

## 背景

channel tiling では overlap 領域の raw trace に複数 tile から予測が出る。従来実装は raw trace ごとに 1 prediction だけを前提に duplicate をエラーとしていたため、overlap を持つ tile 推論に対応できない。

重複 trace は tile 端を弱く、tile 中央を強く使う taper weight で統合する。

## 目的

各 channel tile の予測を raw gather 全体の trace index に戻し、overlap 部を加重平均して、従来と同じ `.coarse.npz` schema で保存する。

## 対象ファイル

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/infer.py`
- `packages/seisai-engine/tests/test_fbpick_coarse.py`

## 実装タスク

- tile merge weight を返す helper を追加する。
  - 例: `_channel_tile_merge_weights(meta, n_traces=...)`
- `channel_tile_count <= 1` または `overlap_traces <= 0` の場合は全 trace weight 1 とする。
- overlap がある場合、tile の左端・右端に ramp weight をかける。
  - gather 先頭 tile の左端は減衰しない。
  - gather 末尾 tile の右端は減衰しない。
  - 中間 seam では両側 tile を taper で混ぜる。
- 従来の duplicate prediction reject を、weighted accumulation に置き換える。
- accumulation 用に以下を持つ。
  - `coarse_pick_weighted_sum`
  - `coarse_pmax_weighted_sum`
  - `coarse_weight_sum`
  - `counts`
- 全 batch 後、`weighted_sum / weight_sum` で gather 全体の prediction を確定する。
- `coarse_pick_i` は round + clip して `int32` に戻す。
- `coarse_pmax` は `float32` に戻す。
- すべての raw trace が 1 回以上 cover されたことを確認する。
- weight が非有限値または 0 以下の場合は fail する。

## 出力契約

`.coarse.npz` の既存 key / shape は変えない。

```text
coarse_pick_i:      (n_traces,)
coarse_pick_t_sec:  (n_traces,)
coarse_pmax:        (n_traces,)
n_traces:           scalar
```

## 非対象

- merge の方式を CLI オプション化することはこの issue では扱わない。
- fine stage / physics stage の変更は行わない。
- model logits の shape 契約は変更しない。

## 受け入れ条件

- overlap trace で duplicate prediction が発生してもエラーにならない。
- overlap trace は weighted average で 1 つの prediction にまとまる。
- 全 raw trace が cover されない場合はエラーになる。
- saved `.coarse.npz` は従来 schema のまま。
- `n_traces` が元 gather 全体の trace 数になる。

## テスト観点

- large gather で model が tile 数ぶん呼ばれること。
- saved `coarse_pick_i.shape == (n_traces,)` であること。
- saved `coarse_pmax.shape == (n_traces,)` であること。
- overlap があるケースで duplicate エラーが発生しないこと。
- small gather の出力が従来どおりであること。

## 依存関係

- Issue 003
- Issue 004
